interfaces/Readme.md

```diff
@@ -0,0 +1,482 @@
+
+        // Now use our SMS service to send the message.
+        if (!this.sms_service.isActive()) {
+            return false; // Our SMS is offline.
+        }
+
+        this.sms_service.sendText(number, message);
+
+        return true;
+
+    }
+
+}
+```
+
+Omitting many details about this class, we see that it implements the `Notifier` interface.
+This means that it is required to define a `send` function with the same signature as the `Notifier` interface.
+Here, we see that we must check whether users have a phone number.
+We then check that our SMS service is actually able to send a text.
+Finally, once these checks pass, we can send out our message.
+
+Now, let's see how the email notifier might be implemented!
+
+```java
+// Sends messages via email.
+class EmailNotifier implements Notifier {
+
+    // ... Constructors, data, and other necessities here.
+
+    @Override
+    public boolean send(User user, String message) {
+
+        String email = user.getEmail();
+        String from_email = "abc@example.com";
+
+        // We must format the email as HTML in order to be sent.
+        formatted = this.formatMessage(message);
+
+        this.email_service.sendEmail(email, from_email, formatted)
+
+        return true;
+
+    }
+
+}
+```
+
+Again, this class implements the `Notifier` and it's `send` method.
+The `EmailNotifier` and `TextNotifier` go about the actual details of sending an message differently.
+Notice the `EmailNotifier` must format it's message (details omitted) and gather the necessary email addresses.
+Whereas the `TextNotifier` had to do additional verification to our user before sending.
+
+Now, why do we want this?
+We can see why this interface is useful by looking at how you, a developer, might leverage this.
+Consider the following usage of the `Notifier` interface where we try to send welcome messages to new users:
+
+```java
+
+class Welcomer {
+
+    void sendWelcomeMessage(User user, Notifier notifier) {
+
+        notifier.send(user, "Welcome to our site!");
+
+    }
+
+}
+
+```
+
+Albeit simple, this illustrates that now our `Welcomer` class (and any other class that must notify users) can do so using only the `Notifier` interface.
+That is, we can do both of the following simultaneously:
+
+```java
+EmailNotifier emailer = EmailNotifier();
+TextNotifier texter = TextNotifier();
+Welcomer welcomer = Welcomer();
+
+welcomer.sendWelcomeMessage(user_1, emailer);
+welcomer.sendWelcomeMessage(user_2, texter);
+```
+
+The first user will receive an email welcome and the second will receive a text message.
+In both cases, the `Welcomer` does not know which implementation of the `Notifier` it is using.
+This means that the `Welcomer` does not _care_ about how someone is notified, just that they can be notified.
+
+## The List Interface
+
+Now we will take a look at a very common interface within Java projects.
+In Python, you have the `List` class that you'd use for most of your data.
```

Comment:
`list` needs to be lowercase

Reply:
Whenever I have something within backticks, I try to always have it be valid code (or at least, verbatim what it refers to). In this case, I'm referring directly to the java class `List` so I am treating it as proper. Would we like to change this?